### PR TITLE
chore: add utility for widget level acl

### PR DESCRIPTION
**Pull request by nsarupr**
#### PR fixes following issue(s)
Fixes https://github.com/appsmithorg/appsmith/issues/23260

#### Type of change
- Chore (housekeeping or task changes that don't impact user perception)

#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Test-plan-implementation#speedbreaker-features-to-consider-for-every-change) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans/_edit#areas-of-interest)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

File: app/client/src/pages/Settings/FormGroup/Checkbox.tsx [hunk 1]
```diff
@@ -82,7 +83,7 @@ const StyledFieldCheckboxGroup = styled.div`
 const formValuesSelector = getFormValues(SETTINGS_FORM_NAME);
 
 export function CheckboxComponent({ setting }: SettingComponentProps) {
-  const settings = useSelector(formValuesSelector);
+  const settings: any = useSelector(formValuesSelector);
```
Comment (dipyamanbiswas07) [inline app/client/src/pages/Settings/FormGroup/Checkbox.tsx:86]: can we add a type here?

File: app/client/src/pages/Settings/SaveSettings.tsx [hunk 2]
```diff
@@ -44,7 +46,12 @@ const saveAdminSettings = (props: SaveAdminSettingsProps) => {
         onClick={onSave}
         size="md"
       >
-        {createMessage(() => "Save & Restart")}
+        {createMessage(() =>
+          updatedTenantSettings?.length === Object.keys(settings).length &&
+          updatedTenantSettings?.length !== 0
+            ? "Save"
```
Comment (dipyamanbiswas07) [inline app/client/src/pages/Settings/SaveSettings.tsx:52]: Lets add from constants?

File: app/client/src/pages/Settings/FormGroup/Checkbox.tsx (hunk 1 above)
Comment (ankitakinger) [reply app/client/src/pages/Settings/FormGroup/Checkbox.tsx]: done

File: app/client/src/pages/Settings/SaveSettings.tsx (hunk 2 above)
Comment (ankitakinger) [reply app/client/src/pages/Settings/SaveSettings.tsx]: done

**Review by dipyamanbiswas07 — DISMISSED** [review]
FE changes LGTM

**Review by trishaanand — DISMISSED** [review]

**Review by dipyamanbiswas07 — APPROVED** [review]
FE changes LGTM

**Pull request merged**